src/main/java/com/coveo/feign/ReflectionErrorDecoder.java

```diff
@@ -169,26 +165,29 @@ private void processDeclaredThrownExceptions(Class<?>[] thrownExceptionsClasses)
   }
 
   private RuntimeException getRuntimeExceptionByReflection(String exceptionKey, T apiResponse)
-      throws InstantiationException, IllegalAccessException, IllegalArgumentException,
-          InvocationTargetException {
+      throws InstantiationException, IllegalAccessException, InvocationTargetException {
     RuntimeException runtimeExceptionToBeThrown =
         runtimeExceptionsThrown.get(exceptionKey).instantiate();
-    detailMessageField.set(runtimeExceptionToBeThrown, getMessageFromResponse(apiResponse));
+    if (runtimeExceptionToBeThrown instanceof ExceptionMessageSetter) {
+      ((ExceptionMessageSetter) runtimeExceptionToBeThrown)
+          .setExceptionMessage(getMessageFromResponse(apiResponse));
+    }
     return runtimeExceptionToBeThrown;
   }
 
   private S getExceptionByReflection(String exceptionKey, T apiResponse)
-      throws IllegalArgumentException, IllegalAccessException, InstantiationException,
-          InvocationTargetException {
+      throws IllegalAccessException, InstantiationException, InvocationTargetException {
     S exceptionToBeThrown = exceptionsThrown.get(exceptionKey).instantiate();
-    detailMessageField.set(exceptionToBeThrown, getMessageFromResponse(apiResponse));
+    if (exceptionToBeThrown instanceof ExceptionMessageSetter) {
+      ((ExceptionMessageSetter) exceptionToBeThrown)
```

Comment:
maybe during initialization?

Reply:
well, it's set to _something_ and depending on how your constructors are set up, it might be the expected message.

I could log something once when parsing the exception, that's not a bad idea I'll try something

Reply:
I've added a log that logs only once.